apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -839,6 +839,35 @@ static void tc_pthread_pthread_key_create_set_getspecific(void)
 }
 
 /**
+* @fn                   :tc_pthread_pthread_key_create_delete
+* @brief                :thread-specific data key creation & deletion
+* @Scenario             :The pthread_key_create() function shall create a thread-specific data key visible to all threads in the process.
+*                        If successful, the pthread_key_create() function shall store the newly created key value.
+*                        otherwise, an error number shall be returned to indicate the error.
+*                        The pthread_key_delete() function shall delete a thread-specific data key previously returned by pthread_key_create()
+*                        If successful, the pthread_key_delete() function shall return zero;
+*                        otherwise, an error number shall be returned to indicate the error.
+* API's covered         :pthread_key_create, pthread_key_delete
+* Preconditions         :pthread_key_create
+* Postconditions        :none
+* @return               :void
+*/
+static void tc_pthread_pthread_key_create_delete(void)
+{
+	int ret_chk;
+	g_tlskey = 0;
+
+	/* Cannot create keys more than PTHREAD_KEYS_MAX, Not able to delete key */
+	ret_chk = pthread_key_create(&g_tlskey, NULL);
+	TC_ASSERT_EQ("pthread_key_create", ret_chk, OK);
+
+	ret_chk = pthread_key_delete(g_tlskey);
```

Comment:
pthread_key_delete() is not implemented yet. so making tc for this api is not adequate.